src/main/asciidoc/_chapters/hbase_mob.adoc

```diff
@@ -181,84 +323,59 @@ suit your environment, and restart or rolling restart the RegionServer.
-dedicated mob data area under the hbase root directory specific to the namespace, table, mob
-logical region, and column family. In general that means a path structured like:
+=== MOB Troubleshooting
 
-----
-%HBase Root Dir%/mobdir/data/%namespace%/%table%/%logical region%/%column family%/
-----
+==== Adjusting the MOB cleaner's tolerance for new hfiles
 
-With default configs, an example table named 'some_table' in the
-default namespace with a MOB enabled column family named 'foo' this HDFS directory would be
+The MOB cleaner chore ignores all MOB hfiles that were created more recently than an hour prior to
+the start of the chore to ensure we don't miss the reference metadata from the corresponding regular
+hfile. Without this safety check it would be possible for the cleaner chore to see a MOB hfile for
+an in progress flush or compaction and prematurely archive the MOB data. This default buffer should
+be sufficient for normal use.
 
-----
-/hbase/mobdir/data/default/some_table/372c1b27e3dc0b56c3a031926e5efbe9/foo/
-----
+You will need to adjust the tolerance if you use write amplification optimized MOB compaction and
+the combination of your underlying filesystem performance and data shape is such that it could take
+more than an hour to complete major compaction of a single region. For example, if your MOB data is
+distributed such that your largest region adds 80GB of MOB data between compactions that include
+rewriting MOB data and your HDFS cluster is only capable of writing 20MB/s for a single file then
+when performing the optimized compaction the Region Server will take about a minute to write the
+first 1GB MOB hfile and then another hour and seven minutes to write the remaining seventy-nine 1GB
+MOB hfiles before finally committing the new reference hfile at the end of the compaction. Given
+this example, you would need a larger tolerance window.
 
-These MOB hfiles are maintained by special chores in the HBase Master rather than by any individual
-Region Server. Specifically those chores take care of enforcing TTLs and compacting them. Note that
-this compaction is primarily a matter of controlling the total number of files in HDFS because our
-operational assumptions for MOB data is that it will seldom update or delete.
+You will also need to adjust the tolerance if Region Server flush operations take longer than an
+hour for the two HDFS move operations needed to commit both the MOB hfile and the normal hfile that
+references it. Such a delay should not happen with a normally configured and healthy HDFS and HBase.
 
-When a given MOB hfile is no longer needed as a result of our compaction process it is archived just
-like any normal hfile. Because the table's mob region is independent of all the normal regions it
-can coexist with them in the regular archive storage area:
+The cleaner's window for "too recent" is controlled by setting `hbase.mob.min.age.archive` to a
+positive integer number of milliseconds.
```

Comment:
This sounds like a very important advice. Could we emphasise the importance of this tuning to avoid data loss using a NOTE] tag?

Also, still thinking on potential operation issues we might need to support, is there any idea of an extra safeguard check that could compare mob compaction time taken with the configured value for _hbase.mob.min.age.archive_ and raise alerts/warnings in case it's higher than the threshold?

Reply:
> This sounds like a very important advice. Could we emphasise the importance of this tuning to avoid data loss using a NOTE] tag?

Where would you suggest such a NOTE go in the current documentation? I could maybe put it in the section on enabling write amplification optimized MOB compaction?

> Also, still thinking on potential operation issues we might need to support, is there any idea of an extra safeguard check that could compare mob compaction time taken with the configured value for hbase.mob.min.age.archive and raise alerts/warnings in case it's higher than the threshold?

We could add that in a follow on jira if you like.

Reply:
added a note to the optimized mode